common/network/tests/test_uri_parser.c

```diff
@@ -0,0 +1,50 @@
+#include <stdio.h>
+#include <unity/unity.h>
+#include "common/network/uri_parser.h"
+
+void test_uri_parser_ipv4(void) {
+  char *uri_ipv4 = "tcp://8.8.8.8:4242";
+  char *scheme;
+  size_t scheme_len = 5;
+  char *host;
+  size_t host_len = 17;
+  uint16_t *port;
+
+  scheme = (char *)malloc(sizeof(char) * scheme_len);
+  host = (char *)malloc(sizeof(char) * host_len);
+  port = (uint16_t *)malloc(sizeof(uint16_t));
```

Comment:
Just use static variables, no need for dynamic allocation here (which you are not freeing btw)